src/python/library/tritonclient/grpc/__init__.py

```diff
@@ -1286,6 +1291,121 @@ def wrapped_callback(call_future):
+            sequence. Default value is False. This argument is ignored if
+            'sequence_id' is 0.
+        priority : int
+            Indicates the priority of the request. Priority value zero
+            indicates that the default priority level should be used
+            (i.e. same behavior as not specifying the priority parameter).
+            Lower value priorities indicate higher priority levels. Thus
+            the highest priority level is indicated by setting the parameter
+            to 1, the next highest is 2, etc. If not provided, the server
+            will handle the request using default setting for the model.
+        timeout : int
+            The timeout value for the request, in microseconds. If the request
+            cannot be completed within the time the server can take a
+            model-specific action such as terminating the request. If not
+            provided, the server will handle the request using default setting
+            for the model.
+        client_timeout : float
+            The maximum end-to-end time, in seconds, the request is allowed
+            to take. The client will abort request and provide
+            error with message "Deadline Exceeded" in the callback when the
+            specified time elapses. The default value is None which means
+            client will wait for the response from the server.
+        headers: dict
+            Optional dictionary specifying additional HTTP
+            headers to include in the request.
+        compression_algorithm : str
+            Optional grpc compression algorithm to be used on client side.
+            Currently supports "deflate", "gzip" and None. By default, no
+            compression is used.
+
+        Raises
+        ------
+        InferenceServerException
+            If server fails to issue inference.
+        """
+        if headers is not None:
+            metadata = headers.items()
+        else:
+            metadata = ()
+
+        if type(model_version) != str:
+            raise_error("model version must be a string")
+
+        request = _get_inference_request(model_name=model_name,
+                                         inputs=inputs,
+                                         model_version=model_version,
+                                         request_id=request_id,
+                                         outputs=outputs,
+                                         sequence_id=sequence_id,
+                                         sequence_start=sequence_start,
+                                         sequence_end=sequence_end,
+                                         priority=priority,
+                                         timeout=timeout)
+        if self._verbose:
+            print("asyncio_infer, metadata {}\n{}".format(metadata, request))
+
+        try:
+            response = await self._aio_client_stub.ModelInfer(
```

Comment:
Does this wait here for the request to complete? That makes asyncio_infer a synchronous function doesn't it? I think we want the user to get back a result object that is really a asyncio future that they can choose to await on however they want. I think that means we may not need a whole new function just an option on the existing async_infer.

Reply:
>Does this wait here for the request to complete? That makes asyncio_infer a synchronous function doesn't it? 

My understanding is that when the await keyword is reached it passes execution to the event loop which gives control to a task with an event that's ready. It means it's a non-blocking call and the method `asyncio_infer` is an awaitable coroutine.

>I think we want the user to get back a result object that is really a asyncio future that they can choose to await on however they want. I think that means we may not need a whole new function just an option on the existing async_infer.

This would also work and could be done like in the email I sent you. I think it's more complicated for the end user though - especially with the different types of futures. Plus, most Python guides advise API owners not to expose futures to end users. I've measured the performance between the implementation in this PR vs the future based one and found this one has a slight edge.